### ISSUE TITLE
[v2 FEATURE]: Add option for adaptive annealing coefficient to dirichlet loss

### ISSUE THREAD

**Issue by KnathanM** [chemprop/chemprop#1068]
In the CLI we set `args.v_kl` to 0 for evidential. This flag is also used for the `v_kl` in dirichlet, but in the dirichlet paper, they suggest using a `v_kl` that changes with epoch:
![image](https://github.com/user-attachments/assets/85a52320-52d9-4243-9ecd-e5ad2844bf50)

I don't really have a good idea how we would add this functionality. Also not totally sure I understand the paper, and the v1 implementation used a constant `v_kl` (there `lam`) with a default of 0.

**Comment by davidegraff** [comment]
IMO you would need to implement this via some custom `Scheduler` class which "`step()`s" its parameter at each training step or each epoch inside the `LightningModule.on_train_{batch,epoch}_end()` methods. High level, these classes could "attach" to certain attributes of their input object and modify the respective value when the scheduler calls `step()`. The abstraction here is that the underlying class has no knowledge that its parameters are being stepped (similar to a pytorch module).
